home/views.py

```diff
@@ -1,81 +1,243 @@
+    first logs in they don't have a tag and they are directed based on the group they belong to with
+    approximately the highest permission level.
+    redirects ref:
+    https://realpython.com/django-redirects/#django-redirects-a-super-simple-example
+    """
+    if request.user.userprofile.change_pwd:
+        return redirect("change_pwd")
+    else:
+        if request.user.groups.all().count() == 0:
+            return redirect("home-register_after_oauth")
+        elif request.user.groups.filter(name="staff").exists():
+            return redirect("staff")
+        elif request.user.groups.filter(name="teacher").exists():
+            return redirect("teacher")
+        elif request.user.groups.filter(name="volunteer").exists():
+            return redirect("volunteer")
+        elif request.user.groups.filter(name="donor").exists():
+            return redirect("donor")
+        else:  # request.user.groups.filter(name = 'student').exists()
+            return redirect("student")
+
+
+@login_required
+def change_pwd(request):
+    if request.method == "POST":
+        form = ChangePwdForm(request.user, request.POST)
+        if form.is_valid():
+            user = form.save()
+            update_session_auth_hash(request, user)  # Important!
+            messages.success(request, "Your password was successfully updated!")
+            request.user.userprofile.change_pwd = False
+            request.user.userprofile.save()
+            return redirect("home-home")
+        else:
+            messages.error(request, "Please correct the error below.")
+    form = ChangePwdForm(request.user)
+    return render(request, "home/change_pwd.html", {"form": form})
+
 
 def logout_view(request):
     logout(request)
-    return render(request, 'home/logout.html')
+    return render(request, "home/logout.html")
+
 
 def login(request):
-    return redirect('login')
+    return redirect("login")
+
 
 def register(request):
-    return render(request, 'home/register.html')
+    return render(request, "home/register.html")
+
 
 def landing_page(request):
-    return render(request, 'home/landing_page.html')
+    return render(request, "home/landing_page.html")
+
 
+@login_required
 def register_after_oauth(request):
-    return render(request, 'home/register_after_oauth.html')
+    """
+    This method catches a user who logs in with no groups, this may happen in two scenarios.
+    Scenario 1: A user logs in with gmail who has never logged in before.  In that situation,
+    the user is required to choose a group to complete registration, and is allowed to proceed.
+    Scenario 2: A user logs in with gmail who has an account already, but has never logged in
+    with gmail.  In that situation, our API auto-generates a new user account, creating a
+    duplicate entry.  This method takes care of that by adding the new user to the old user's
+    groups, and deleting the old user.
+    """
```

Comment:
I think that this approach is a bit dangerous, but can be addressed outside of this PR in a follow-up issue (#11) as we don't have any existing deployments so merging is not so dangerous.